### PR TITLE
Add hasLabel check for hidden inputs

### LINKED ISSUE #106: Issue detecting "hidden" input
I was doing a little proof of concept with tarsier and ran it on https://todomvc.com/examples/react/dist/. They have a checkbox input which they make `opacity: 0`, they then use a background image on the input label to have a custom checkbox style. Screenshots below should help clarify the example. 

Screenshot 2024-07-11 at 3 24 43 PM
Screenshot 2024-07-11 at 3 40 28 PM

I'll admit it's probably not a great example in terms of accessibility but unfortunately many sites are built this way. Anyway this issue is to ask if this has come up before and if anyone had an idea of how to approach it? 

My guess is that (especially for image mode) it would be more reliable to still tag "hidden" inputs. 

For now to get the agent able to mark todos as complete I have just commented out the [isTransparent flag](https://github.com/reworkd/tarsier/blob/main/tarsier/tag_utils.ts#L19).

### PR DESCRIPTION
**Pull request by craigmulligan**
Fixes #106

Current tagging: 

image

With fix:

image

### CODE REVIEW

**Comment by awtkns** [comment]
Hey @craigmulligan thanks for this! Curious what you have been using tarsier for so far :)

**Comment by awtkns** [comment]
Seems to be some conflicts from https://github.com/reworkd/tarsier/pull/119

File: tarsier/tag_utils.ts [hunk 1]
```diff
@@ -237,6 +237,38 @@ function getAllElementsInAllFrames(): HTMLElement[] {
...
 }
 
+function hasLabel(element: HTMLElement): boolean {
+  const tagNames = ["input", "textarea", "select", "button"];
+
+  if (!tagNames.includes(element.tagName.toLowerCase())) {
+    return false;
+  }
+
+  const label = document.querySelector(`label[for='${element.id}']`);
```
Comment (seanmcguire12) [inline tarsier/tag_utils.ts:247]: Should make sure we are escaping any special characters in the id
```suggestion
  const escapedId = CSS.escape(element.id);
  const label = document.querySelector(`label[for="${escapedId}"]`);
```

**Comment by asim-shrestha** [comment]
@craigmulligan moved logic around to be a bit more consistent

**Review by asim-shrestha — APPROVED** [review]

**Pull request merged**